packages/db/src/models/granule.ts

```diff
@@ -5,30 +5,79 @@ import { tableNames } from '../tables';
 
 import { PostgresGranule, PostgresGranuleRecord } from '../types/granule';
 
+import { GranuleExecutionHistoryPgModel } from './granule-execution-history';
+
 export default class GranulePgModel extends BasePgModel<PostgresGranule, PostgresGranuleRecord> {
   constructor() {
     super({
       tableName: tableNames.granules,
     });
   }
 
-  upsert(
+  async createWithExecutionHistory(
+    knexOrTrx: Knex | Knex.Transaction,
+    item: PostgresGranule,
+    executionCumulusId: number,
+    granuleExecutionHistoryPgModel = new GranuleExecutionHistoryPgModel()
+  ) {
+    const [granuleCumulusId] = await this.create(knexOrTrx, item);
+    await granuleExecutionHistoryPgModel.create(
+      knexOrTrx,
+      {
+        granule_cumulus_id: granuleCumulusId,
+        execution_cumulus_id: executionCumulusId,
+      }
+    );
+    return [granuleCumulusId];
+  }
+
+  getWithExecutionHistory(
```

Comment:
It might be premature to add this method - we're not using it anywhere yet. Nor will we need to until we want the API/whatever to actually return all of the executions for a granule. So I could just remove this, though it seems ideal to write it now while this is top of mind. But I wanted to get another opinion

Reply:
Yeah, the other issue is that I'm assuming two things:

- We want to get the granules and their executions in one query, as opposed to two
- We want to return the executions for a granule as an aggregation of their cumulus IDs

Neither of which may turn out to be true depending on how we want to fetch this data later. So I'm on the fence here

Reply:
I'm going to say YAGNI for now and remove it

Reply:
5440e0792e8e84bc2db1ee52f72907f27e99cbec